### PR TITLE
Refactor API client implementations

### LINKED ISSUE #3489: Refactor API client implementations to be based on new API client
Follow up of #3488

The new client should be the pattern of all clients, including those from `SOAPServer` and/or `StUFService`.

Additionally, check if we use requests-oauth2client (indirectly) and ensure that the correct `session` instance is provided.

**Checklist**

- [x] Create adapter for SOAPService
- [x] Create adapter for StUFService
- [x] Check requests-oauth2client usage
    - [x] `openforms.contrib.microsoft`
- [x] Check any other client definitions (zeep transport supports session kwarg)
    - [x] `openforms.contrib.bag` (move this into kadaster!)
    - [x] `openforms.contrib.brp`
    - [x] `openforms.contrib.camunda`
    - [x] `openforms.contrib.kadaster`
    - [x] `openforms.contrib.kvk`
    - [x] Haal Centraal / BRP (pre-request/token exchange thing is relevant here!)
    - [x] zgw
- [x] Figure out how to handle exception suppressing while allowing config checks to "crash"
- [ ] Update docs
    - [ ]  Developer docs for api clients
    - [ ] BAG configuration?
    - [ ] Kadaster configuration -> update model field labels
- [x] Check extensions
    - [x] Token exchange
    - [x] [Haal Centraal HR Prefill](https://github.com/open-formulieren/open-forms-ext-haalcentraal-hr)
- [x] Ensure zgw-consumers NLX integration still works (look into `Session.send`, `Session.prepare_request` and response hooks)
    - [x] updating base/root URL
    - [x] rewriting response data / params / headers

ZGW/ZDS-client is beast by itself and probably best left for a separate PR 😬

### PR DESCRIPTION
**Pull request by sergei-maertens**
Partly closes #3489

### CODE REVIEW

File: src/openforms/contrib/kvk/models.py [hunk 1]
```diff
@@ -39,23 +34,3 @@ class KVKConfig(SingletonModel):
 
     class Meta:
         verbose_name = _("KvK configuration")
-
-    # FIXME: Patch ZGW Consumer behaviour to allow for 2 KvK API's.
```
Comment (sergei-maertens) [inline src/openforms/contrib/kvk/models.py:43]: HURAY

**Comment by sergei-maertens** [comment]
tomorrow: sort out coverage so patch coverage is 100%

File: src/openforms/contrib/kadaster/config_check.py [hunk 2]
```diff
@@ -39,3 +34,34 @@ def get_config_actions() -> list[Action]:
...
+                _("Invalid response: {exception}").format(exception=exc)
+            ) from exc
+
+    @staticmethod
+    def get_config_actions() -> list[Action]:
+        return [
+            (
+                gettext("Configuration"),
+                reverse(
+                    "admin:bag_bagconfig_change",
```
Comment (SilviaAmAm) [inline src/openforms/contrib/kadaster/config_check.py:63]: I get in the admin:
```
Interne fout: Reverse for 'bag_bagconfig_change' not found. 'bag_bagconfig_change' is not a valid view function or pattern name. 
```

Comment (sergei-maertens) [reply src/openforms/contrib/kadaster/config_check.py:63]: ah right, that one is gone indeed! weird that it didn't show for :thinking:

**Pull request merged**